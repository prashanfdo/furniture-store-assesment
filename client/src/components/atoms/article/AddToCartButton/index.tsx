import { ButtonHTMLAttributes } from "react";

type AddToCartButtonProps = Pick<
  React.DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
  "onClick"
>;
const AddToCartButton: React.FC<AddToCartButtonProps> = ({ onClick }: AddToCartButtonProps) => {
  return (
    <button
      onClick={onClick}
      className="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
    >
      Add To Cart
    </button>
  );
};

export default AddToCartButton;
