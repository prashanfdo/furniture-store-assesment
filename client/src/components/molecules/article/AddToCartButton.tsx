import styled from "@emotion/styled";
import { Article } from "types";

const ButtonContainer = styled.button`
  text-align: center;
`;

type AddToCartButtonProps = {
  product: Article;
};
const AddToCartButton: React.FC<AddToCartButtonProps> = ({ product }: AddToCartButtonProps) => {
  return <ButtonContainer>Add To Cart</ButtonContainer>;
};

export default AddToCartButton;
