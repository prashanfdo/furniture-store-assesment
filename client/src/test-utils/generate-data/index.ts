export * from "./article";
export * from "./category";
export * from "./search";
