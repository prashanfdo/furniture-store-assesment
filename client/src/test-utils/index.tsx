import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
