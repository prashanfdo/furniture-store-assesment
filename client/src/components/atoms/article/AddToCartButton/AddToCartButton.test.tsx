import { render } from "test-utils";
import AddToCartButton from "./index";

describe("Components", () => {
  describe("Atoms", () => {
    describe("AddToCartButton", () => {
      it("should render", () => {
        const { container } = render(<AddToCartButton />);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
