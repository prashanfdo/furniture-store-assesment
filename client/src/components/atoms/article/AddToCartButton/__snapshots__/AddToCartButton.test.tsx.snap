// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Atoms AddToCartButton should render 1`] = `
<div>
  <button
    class="block w-full p-2 mt-2 text-center text-gray-600 bg-white border border-solid rounded cursor-pointer border-coral-200 hover:border-coral-600 hover:bg-coral-600 hover:text-white"
  >
    Add To Cart
  </button>
</div>
`;
